packages/core/src/actions.ts

```diff
@@ -543,7 +543,9 @@ export function pure<TContext, TEvent extends EventObject>(
   getActions: (
     context: TContext,
     event: TEvent
-  ) => SingleOrArray<ActionObject<TContext, TEvent>> | undefined
+  ) =>
+    | SingleOrArray<ActionObject<TContext, TEvent> | TEvent['type']>
```

Comment:
This is incorrect - you are saying here that it's allowed to return the **event**'s type as an action type. This should allow returning `TAction['type']`

Reply:
Wow that's a major derp moment 😅 Good call

Reply:
Switched it to `ActionObject<TContext, TEvent>['type']` since we don't have `TAction` in this case

Reply:
In such a case - you could simplify this to just `string` but perhaps this gives more context behind what that `string` is.